src/main/scala/com.snowplowanalytics.snowplow.micro/Routing.scala

```diff
@@ -39,41 +43,61 @@ private[micro] object Routing {
   def getMicroRoutes(
       collectorConf: CollectorConfig,
       collectorSinks: CollectorSinks
-  ): Route = {
+  )(implicit ec: ExecutionContext): Route = {
+    val c = new CollectorService(collectorConf, collectorSinks)
     val collectorRoutes = new CollectorRoute {
-      override def collectorService =
-        new CollectorService(collectorConf, collectorSinks)
+      override def collectorService = c
     }.collectorRoute
 
-    val microRoutes: Route = pathPrefix("micro") {
-      (get | post) {
-        path("all") {
-          complete(ValidationCache.getSummary())
-        } ~ path("reset") {
-          ValidationCache.reset()
-          complete(ValidationCache.getSummary())
-        }
-      } ~ get {
-        path("good") {
-          complete(ValidationCache.filterGood(FiltersGood(None, None, None, None)))
-        } ~ path("bad") {
-          complete(ValidationCache.filterBad(FiltersBad(None, None, None)))
-        }
-      } ~ post {
-        path("good") {
-          entity(as[FiltersGood]) { filters =>
-            complete(ValidationCache.filterGood(filters))
+    withCors(c) {
+      pathPrefix("micro") {
+        (get | post) {
+          path("all") {
+            complete(ValidationCache.getSummary())
+          } ~ path("reset") {
+            ValidationCache.reset()
+            complete(ValidationCache.getSummary())
           }
-        } ~ path("bad") {
-          entity(as[FiltersBad]) { filters =>
-            complete(ValidationCache.filterBad(filters))
+        } ~ get {
+          path("good") {
+            complete(ValidationCache.filterGood(FiltersGood(None, None, None, None)))
+          } ~ path("bad") {
+            complete(ValidationCache.filterBad(FiltersBad(None, None, None)))
           }
-        } 
-      } ~ {
-        complete("Path for micro has to be one of: /all /good /bad /reset")
+        } ~ post {
+          path("good") {
+            entity(as[FiltersGood]) { filters =>
+              complete(ValidationCache.filterGood(filters))
+            }
+          } ~ path("bad") {
+            entity(as[FiltersBad]) { filters =>
+              complete(ValidationCache.filterBad(filters))
+            }
+          } 
+        } ~ options {
+          complete(StatusCodes.OK)
+        } ~ {
+          complete("Path for micro has to be one of: /all /good /bad /reset")
+        }
       }
-    }
-
-    microRoutes ~ collectorRoutes
+    } ~ collectorRoutes
   }
+
+  /** Wrap a Route with CORS header handling.
+  *
+  *  Reuses the implementation used by the stream collector
+  */
+  private def withCors(c: CollectorService)(route: Route)(implicit ec: ExecutionContext): Route =
```

Comment:
Out of curiosity what needs the `ExecutionContext` in this function ?

Reply:
IIRC, extracting the `requestContext` seems to require this.

Reply:
It's because `route(requestContext)` returns a `Future`.  The execution context is needed [for the `.map` on the future](https://www.scala-lang.org/api/2.12.2/scala/concurrent/Future.html#map[S](f:T=%3ES)(implicitexecutor:scala.concurrent.ExecutionContext):scala.concurrent.Future[S]).